lib/rules/display-name.js

```diff
@@ -191,10 +206,13 @@ module.exports = Components.detect(function(context, components, utils) {
       var list = components.list();
       // Report missing display name for all components
       for (var component in list) {
-        if (!list.hasOwnProperty(component) || list[component].hasDisplayName) {
+        if (forbidExplicitDefinition && list[component].hasDisplayName) {
```

Comment:
The `hasDisplayName` property is `true` if we find an explicit displayName or if the transpiler can find one.

I think we should differentiate the 2 "types" of names and only warn if this is an explicit displayName.